service/src/main/java/bio/terra/pipelines/service/PipelineRunsService.java

```diff
@@ -93,21 +94,31 @@ public Map<String, Map<String, String>> preparePipelineRun(
       throw new InternalServerErrorException("%s workspaceId not defined".formatted(pipelineName));
     }
 
+    if (getPipelineRun(jobId, userId) != null) {
```

Comment:
note that this only throws if a jobId already exists that this user submitted. if another user had already submitted a job with this jobId, this currently won't trigger an exception until we attempt to write the job to the db, at which point it will throw a DuplicateObjectException (still a 400 Bad Request). 

should we instead check for any jobId matching this one? in that case the calling user wouldn't be able to retrieve details for it. 

the existing behavior is fine, but it is inefficient in the case that the jobId does exist from another user's request, in that we'll call WSM to retrieve SAS urls before running into the error at db write time.